sstables/sstable_directory.cc

```diff
@@ -651,7 +651,7 @@ future<sstring> sstable_directory::create_pending_deletion_log(opened_directory&
         try {
             auto trim_size = base_dir.native().size() + 1; // Account for the '/' delimiter
             for (const auto& sst : ssts) {
-                auto toc = sst->toc_filename();
+                sstring toc = fmt::to_string(sst->toc_filename());
```

Comment:
this creates a temporary `std::string`, and throws it away, after copying it to `sstring`. how about using `seastar::to_sstring()` if we really want to use `sstring` here.

Reply:
Hm... I was surprised to see that std::string -> sstring conversion comes via `basic_sstring(const? std::basic_string<char_type>& x)` constructor which does copy things. I'll apply your suggestion, but why don't we have move-assignment operator for such cases, it seems common enough?